docs/generateDocs.py

```diff
@@ -0,0 +1,203 @@
+    if an annotation is found then finds the function signature
+    """
+    annotation = lines[i].strip()
+    index = annotation.find("(")
+    #annotation was found
+    if (index != -1):
+        if annotation[1:index].lower() in annotation_filter:
+            #regex = @give(...)
+            annotation = escape_regex(annotation)
+            #looking for signature
+            signature = None
+            signature_incomplete = True
+            while signature_incomplete:
+                i += 1
+                line = lines[i].strip()
+                if line.startswith("public"):
+                    signature = line
+                    signature_incomplete = not signature.endswith("{")
+                elif signature:
+                    #Fixes incomplete signatures when they are longer than one line
+                    signature = signature + line
+                    signature_incomplete = not signature.endswith("{")
+            signature = signature.replace("{", "")
+            level = 1
+            starting_line_num = i
+            i += 1
+            steps.append(Annotation(file_name, starting_line_num, annotation, signature))
+            return i
+    return i
+
+def scan_file(file_name):
+    if not file_name.endswith(".java"):
+        return
+    try:
+        with open(file_name, "r") as f:
+            #Goes through all lines and tries to find annotations
+            line_num = 0
+            lines = f.readlines()
+            while line_num < len(lines):
+                line = lines[line_num].strip()
+                if line.startswith("@"):
+                    line_num = process_lines(file_name, lines, line_num)
+                line_num += 1
+    except Exception as e:
+        print(e)
+
+def create_index(paths):
+    with open(OUTPUTDIR + "index.md", "w") as f:
+        f.write("# Documentation of defined steps\n")
+        for p in paths:
+            #Links all other files to index
+            name = p.replace("-", " ").capitalize()
+            p = p + ".md"
+            if os.path.exists(OUTPUTDIR + p):
+                f.write("* [{}]({})\n".format(name, p))
+        f.write("## All defined steps\n")
+        write_table_header(f, all_steps)
+        for s in all_steps:
+            write_step_to_file(f, s)
+    print("generated: index.md")
+
+def write_table_header(f, steps):
+    if STYLE == "file":
+        f.write("Total defined steps: " + str(len(steps)) + "\n\n")
+        f.write("| Definition | File name | Line number |\n")
+        f.write("| --- | --- | --- |\n")
+    elif STYLE == "ansig":
+        f.write("Total defined steps: " + str(len(steps)) + "\n\n")
+        f.write("| Definition | Signature |\n")
+        f.write("| --- | --- |\n")
+    elif STYLE == "compact":
+        f.write("Total defined steps: " + str(len(steps)) + "\n\n")
+        f.write("| Definition & Signature |\n")
+        f.write("| --- |\n")
+
+def write_step_to_file(f, s):
```

Comment:
Have you considered using a templating engine such as jinja?

Reply:
Not really, I feel like it's overkill for this usage... but it would make the initial setting up more comfortable probably :D